analysis/reports/generator.py

```diff
@@ -68,3 +76,64 @@ def build_pdf_report(self) -> str:
             verbose=True,
         )
         return self.output_pdf
+
+    def build_pdf_charts(self):
+
+        output_pdf = PdfPages(
+            os.path.join(
+                self.output_dir, "generated_charts", self.filename_base + "_charts" + ".pdf"
```

Comment:
Does `generated_charts` will need to be created prior?

Reply:
can you explain more on this?

Reply:
I pulled your branch and when I ran it I got an error that `generated_charts` did not exist. I had to create it manually for the code to run successfully.

Reply:
thanks for catching that!  I added a check for the folder and create if not exists.